agent/agent.go

```diff
@@ -754,6 +755,12 @@ func (a *Agent) Start(ctx context.Context) error {
 			return fmt.Errorf("can't start agent: client agents are not supported with v2 resources")
 		}
 
+		// the conn is used to connect to the consul server agent
+		conn, err := a.baseDeps.GRPCConnPool.ClientConn(a.baseDeps.RuntimeConfig.Datacenter)
```

Comment:
This seems unsafe, because it's attempting to use a single connection that is established up-front indefinitely from what I can tell. Shouldn't we be providing the connection pool object to this server so that it can re-dial, as needed?

Reply:
Thanks for your review @hashi-derek!
From what I found in the code base, when we dial to the datacenter, we actually dial to the servers: [implementation(`server` type is hardcoded)](https://github.com/hashicorp/consul/blob/main/agent/grpc-internal/client.go#L225) and in the [BaseDeps setup](https://github.com/hashicorp/consul/blob/main/agent/setup.go#L220) the agent used that implementation.

Reply:
Yeah, it looks like I was mistaken. There's some more magic going on under the hood with that connection than I thought. The file `agent/grpc-internal/balancer/balancer.go` has a little bit more information about what is going on.